### PR TITLE
enhance storage-opt validation logic in overlay2 driver

### PR DESCRIPTION
**Pull request by mountkin**
Signed-off-by: Shijiang Wei <mountkin@gmail.com>

**- What I did**
- fix a potential panic in the `CreateReadWrite` function in overlay2 driver
- don't add `"size": "0"` to `opts.StorageOpt` map if no daemon options configured

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

eagle
 (An eagle seeking help when Mangkhut struck Shenzhen, China)

### CODE REVIEW

**Comment by dmcgowan** [comment]
To be considered a bug, what options did you give and what was the expected result? There are still no storage opts supported on ext4, is this a precursor to either support more storage options or quota on ext4?

File: daemon/graphdriver/overlay2/overlay.go [hunk 1]
```diff
@@ -340,23 +340,23 @@ func (d *Driver) Cleanup() error {
...
+			StorageOpt: make(map[string]string),
 		}
+	} else if opts.StorageOpt == nil {
+		opts.StorageOpt = make(map[string]string)
 	}
 
-	if _, ok := opts.StorageOpt["size"]; !ok {
-		if opts.StorageOpt == nil {
-			opts.StorageOpt = map[string]string{}
-		}
```
Comment (mountkin) [inline daemon/graphdriver/overlay2/overlay.go]: The outer if may cause a panic when `opts.StorageOpt` is nil. I fixed the problem in this PR.

**Comment by mountkin** [comment]
@dmcgowan I'm sorry but I couldn't reproduce the problem any more.
 I remembered that I didn't pass any extra args to dockerd other than `--storage-driver=overlay2`.  And I didn't pass any `storage-opt` to `docker run`.  The error message was `--storage-opt is supported only for overlay over xfs with 'pquota' mount option`
I'll modify the commit log and push again

**Comment by crosbymichael** [comment]
LGTM

**Comment by AkihiroSuda** [comment]
@mountkin Could you rebase for retriggering CI?

**Comment by mountkin** [comment]
@AkihiroSuda rebased

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**